cloudinary-core/src/main/java/com/cloudinary/Url.java

```diff
@@ -357,7 +363,8 @@ public String generate(String source) {
         source = finalizedSource[0];
         String sourceToSign = finalizedSource[1];
 
-        if (sourceToSign.contains("/") && !StringUtils.hasVersionString(sourceToSign) && !httpSource && StringUtils.isEmpty(version)) {
+        if (!excludeVersion && sourceToSign.contains("/") && !StringUtils.hasVersionString(sourceToSign) &&
```

Comment:
@nitzanj, that logic says that if I set a version && I ask to excludeVersion, I still get a version in the URL

Reply:
Setting a version and then requesting to ignore it is undefined behaviour, what do you want to happen here? It's not mentioned in the spec.